cornflow/schemas/execution.py

```diff
@@ -35,6 +36,9 @@ class ExecutionSchema(Schema):
     log_text = fields.Str(dump_only=True)
     log_json = fields.Nested(LogSchema, dump_only=True)
     finished = fields.Boolean(required=False)
+    state = fields.Int(validate=validate.Range(min=MIN_EXECUTION_STATUS_CODE, max=MAX_EXECUTION_STATUS_CODE),
+                       required=False)
+    state_message = fields.Str(required=False)
     created_at = fields.DateTime(dump_only=True)
```

Comment:
So @pchtsp here you commented that It could be complicated.

I'll explain how it is now and I can change it if you want.

Codes are defined in shared/const.py file. From that file are read automatically in the models/execution.py and here as well. If you where to add a new code and modify the min and max code, here the validation in the schema would work.

But, this is setup as well as a database constraint in the corresponding migration file. If you where going to add new codes you would have to create a manual migration (create the file by hand) deleting the constraint and creating it again with the new values. This part is the one that seems more complicated (IMHO) and maybe is the one that we should delete.

Tell me what you think.